discretize/base/base_mesh.py

```diff
@@ -33,103 +31,164 @@ class BaseMesh(properties.HasProperties, InterfaceMixins):
         "nFz": "n_faces_z",
         "nF": "n_faces",
         "vnF": "n_faces_per_direction",
+        "vnC": "shape_cells"
     }
-
-    # Properties
-    _n = properties.Tuple(
-        "Tuple of number of cells in each direction (dim, )",
-        prop=properties.Integer(
-            "Number of cells along a particular direction", cast=True, min=1
-        ),
-        min_length=1,
-        max_length=3,
-        coerce=True,
-        required=True,
-    )
-
-    origin = properties.Array(
-        "origin of the mesh (dim, )",
-        dtype=(float, int),
-        shape=("*",),
-        required=True,
-    )
+    _items = {'shape_cells', 'origin', 'orientation', 'reference_system'}
 
     # Instantiate the class
-    def __init__(self, n=None, origin=None, **kwargs):
-        if n is not None:
-            self._n = n  # number of dimensions
-
+    def __init__(self, shape_cells, origin=None, orientation=None, reference_system=None, **kwargs):
+        if "n" in kwargs:
+            shape_cells = kwargs.pop("n")
         if "x0" in kwargs:
             origin = kwargs.pop('x0')
+        axis_u = kwargs.pop("axis_u", None)
+        axis_v = kwargs.pop("axis_v", None)
+        axis_w = kwargs.pop("axis_w", None)
+        if axis_u is not None and axis_v is not None and axis_w is not None:
+            orientation = np.array([axis_u, axis_v, axis_w])
+
+        shape_cells = tuple((int(val) for val in shape_cells))
+        self._shape_cells = shape_cells
+        # some default values
         if origin is None:
-            self.origin = np.zeros(len(self._n))
-        else:
-            self.origin = origin
+            origin = np.zeros(self.dim)
+        self.origin = origin
+
+        if orientation is None:
+            orientation = Identity()
 
-        super(BaseMesh, self).__init__(**kwargs)
+        self.orientation = orientation
+        if reference_system is None:
+            reference_system = "cartesian"
+        self.reference_system = reference_system
+        super().__init__(**kwargs)
```

Comment:
What is the purpose of this line, `super().__init__(**kwargs)`?.

`BaseMesh` does not inherit any class, so I think this line can be deleted.

It is this line that causes my test with emg3d to fail. If I remove it everything works fine. It can of course be that I use inheritance wrongly in emg3d... but it appears strange to me to have a `super()` call when we are at the very bottom.

Reply:
Ah, now I see. It is in the code because before we inherited from `properties.HasProperties` and `InterfaceMixins` - but now we do not inherit anything any longer, so it should go (I think).

Reply:
I'll put it up on the slack, but the purpose here is if you have multiple inheritance. If a derived class has multiple parents, the next super() call will be to the second parent. I'd have to look at how your doing this in your classes, as it could either be a use-case I hadn't thought of regarding backwards-compatibility, or you're not popping off keyword arguments when initializing parents.

Reply:
So looking at your `TensorMesh` class in emg3d (https://github.com/emsig/emg3d/blob/1a3add15692492af5d9f007b7aac5cd83586289a/emg3d/meshes.py#L188), this `super()` call on `BaseMesh` ends up attempting to call the `__init__` method of `_TensorMesh`, which is probably not what you wanted. It would seem like its purpose is to be a fallback if discretize is not available, and it doesn't need to actually be inherited. It might be best to actually just conditionally define your `TensorMesh`, i.e. instead of defining an empty class at https://github.com/emsig/emg3d/blob/1a3add15692492af5d9f007b7aac5cd83586289a/emg3d/meshes.py#L45, define your own `dTensorMesh` there.

Reply:
Yes, I could change
```
class TensorMesh(dTensorMesh, _TensorMesh):
```
to
```
class TensorMesh(dTensorMesh if dTensorMesh else _TensorMesh):
```

But I think this is a work-around for something introduced here that I believe should not be. IMHO, there should be no `super()` here.